### PR TITLE
[MM-31929] server: add mlog.Error review feautre

### PR DESCRIPTION
**Pull request by isacikgoz**
#### Summary
Mattermod will remind us to review our logging guidelines.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-31929

### CODE REVIEW

File: server/review.go [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+	"strings"
+
+	"github.com/google/go-github/v33/github"
+	"github.com/sourcegraph/go-diff/diff"
+
+	"github.com/mattermost/mattermost-mattermod/model"
+	"github.com/mattermost/mattermost-server/v5/mlog"
+)
+
+const mlogReviewCommentBody = "Gentle reminder to check our logging [principles](https://docs.google.com/document/d/160-9Qfk-1a2_gOU-VHfZ5CbUdkY8BZxq-2VL7XYwqGA) before merging this change."
```
Comment (isacikgoz) [inline server/review.go:20]: I will update this after the https://mattermost.atlassian.net/browse/MM-32010 Also I'm open to better sentences 😅

Comment (agnivade) [reply server/review.go:20]: Yeah, we should add the contents of the document to this section: https://github.com/mattermost/mattermost-developer-documentation/pull/751. Shall we wait until the PR is merged?

Comment (isacikgoz) [reply server/review.go:20]: Deal 🤝

Comment (agnivade) [reply server/review.go:20]: Actually, that PR will take some time. We can go ahead with this pointing to the google doc for now if you prefer.

File: server/review.go [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+				}
+			}
+		}
+	}
+
+	return nil
+}
+
+func getRawDiff(url string) ([]byte, error) {
+	resp, err := http.Get(url) //nolint: gosec
```
Comment (agnivade) [inline server/review.go:77]: We should definitely pass the context from the upper layer in this HTTP call. We don't want to keep it stuck forever.

File: server/review.go [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+							},
+						},
+					}
+
+					_, _, err := s.GithubClient.PullRequests.CreateReview(ctx, pr.RepoOwner, pr.RepoName, pr.Number, review)
+					if err != nil {
+						return fmt.Errorf("could not create the review for PR %s/%s#%d: %w", pr.RepoOwner, pr.RepoName, pr.Number, err)
+					}
+
+					return nil
```
Comment (agnivade) [inline server/review.go:67]: Does this mean that we will be only commenting once even if the PR contains multiple lines of `mlog.Error/Critical`?

File: server/review.go [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+	// We simply iterate every file in the diff and check if the PR includes an mlog.Error/mlog.Critical call
+	for _, fileDiff := range fileDiffs {
+		var position int32 // reset position since it's file specific
+
+		for _, hunk := range fileDiff.Hunks {
+			for _, line := range strings.Split(string(hunk.Body), "\n") {
+				position++
+
+				if len(line) == 0 || line[0] != '+' {
```
Comment (agnivade) [inline server/review.go:43]: line == ""

File: server/review.go [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+
+func (s *Server) reviewMlog(ctx context.Context, pr *model.PullRequest, nodeID, diffURL string) error {
+	b, err := getRawDiff(diffURL)
+	if err != nil {
+		return fmt.Errorf("could not retrieve diff: %w", err)
+	}
+
+	fileDiffs, err := diff.ParseMultiFileDiff(b)
+	if err != nil {
+		return err
```
Comment (agnivade) [inline server/review.go:30]: Maybe worth adding some error context here.

File: server/review.go (hunk 3 above)
Comment (isacikgoz) [reply server/review.go:67]: Yes, I think it will be enough to point out the guideline.

File: server/review.go (hunk 4 above)
Comment (isacikgoz) [reply server/review.go:43]: 🤦

**Review by agnivade — APPROVED** [review]

**Review by nronas — APPROVED** [review]
LGTM 🚢 - it

**Comment by agnivade** [comment]
@isacikgoz - What is this waiting for?

**Comment by isacikgoz** [comment]
@agnivade This is waiting for the docs PR to be merged. 😬

**Pull request merged**